src/libtilt/projection/project_fourier.py

```diff
@@ -92,7 +76,8 @@ def extract_central_slices_rfft(
 
     # flip coordinates in redundant half transform
     conjugate_mask = grid[..., 2] < 0
-    conjugate_mask = einops.repeat(conjugate_mask, '... -> ... 3')
+    # conjugate_mask = einops.repeat(conjugate_mask, '... -> ... 3')
+    conjugate_mask.unsqueeze(-1).repeat(1, 1, 1, 3)
```

Comment:
repeat should be creating a view here and shouldn't be memory intensive even though the tensor is huge - is this not the case?

Reply:
If I am correct, is expand and not repeat the one that is a view. This change was again a requirement for the compiler.

Reply:
I think you're right for the torch API but einops it creates a view where possible - regardless, compilation is super important.

I'm a little hesitant to lose the rank polymorphism here and it looks like this unsqueeze/repeat is specific to `b h w 3` rather than `... h w 3` -> could you try adding some code to intepret the current shape and unsqueeze/repeat according to that? This should allow us to maintain the current flexibility and have compatibility with the compiler

Reply:
This should work `conjugate_mask.unsqueeze(-1).expand(*[-1] * len(conjugate_mask.shape), 3)` and being more memory efficient, since it is a view.